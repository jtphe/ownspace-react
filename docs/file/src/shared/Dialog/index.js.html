<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../../">
  <title data-ice="title">src/shared/Dialog/index.js</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
</head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./">Home</a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
</header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#api">api</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-api">api</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-document">components/Document</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-BackFolder">BackFolder</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Folder">Folder</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-UploadDocument">UploadDocument</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-home">components/Home</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-CreateFile">CreateFile</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Home">Home</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-login">components/Login</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-GenerateTotp">GenerateTotp</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Login">Login</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-SecondAuthScreen">SecondAuthScreen</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TotpAuthScreen">TotpAuthScreen</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-menu">components/Menu</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-IconMenu">IconMenu</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-NavMenu">NavMenu</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#components-userprofile">components/UserProfile</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-PasswordModal">PasswordModal</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#constants">constants</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-RECURRING_ROLE">RECURRING_ROLE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_ONE">ROLE_ONE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_THREE">ROLE_THREE</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-ROLE_TWO">ROLE_TWO</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#router">router</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-AppRouter">AppRouter</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-avatar">shared/Avatar</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Avatar">Avatar</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-clienttext">shared/ClientText</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TextClient">TextClient</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-custombutton">shared/CustomButton</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-CustomButton">CustomButton</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-dialog">shared/Dialog</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-OwnSpaceDialog">OwnSpaceDialog</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-header">shared/Header</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Header">Header</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-WithBack">WithBack</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-logo">shared/Logo</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Logo">Logo</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#shared-text">shared/Text</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TextHelvetica">TextHelvetica</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-openFile">openFile</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-requestAndroidWriteExternalStorage">requestAndroidWriteExternalStorage</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-octetToMoConverter">octetToMoConverter</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-formatDate">formatDate</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-decryptPassword">decryptPassword</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-hashPassword">hashPassword</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-showToast">showToast</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/shared/Dialog/index.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import React from &apos;react&apos;;
import { TextInput, StyleSheet } from &apos;react-native&apos;;
import Text from &apos;@shared/ClientText&apos;;
import { Portal, Dialog, Button } from &apos;react-native-paper&apos;;
import i18n from &apos;@i18n/i18n&apos;;
import PropTypes from &apos;prop-types&apos;;

/**
 * The OwnSpaceDialog component
 * @param {boolean} visible - Visibility of the dialog
 * @param {function} hide - Hide the dialog onDismiss
 * @param {string} dialogTitle - Text for the dialog title
 * @param {string} dialogPlaceholder - Text for the dialog placeholder
 * @param {string} name - Name
 * @param {function} setName - Set the name
 * @param {function} create - Start the function associated to the create button
 */
const OwnSpaceDialog = ({
  visible,
  hide,
  dialogTitle,
  dialogPlaceholder,
  name,
  setName,
  create
}) =&gt; {
  /**
   * Render the OwnSpaceDialog component
   * @returns {React.Component} - OwnSpaceDialog component
   */
  return (
    &lt;Portal&gt;
      &lt;Dialog visible={visible} onDismiss={hide}&gt;
        &lt;Dialog.Title&gt;
          &lt;Text&gt;{dialogTitle}&lt;/Text&gt;
        &lt;/Dialog.Title&gt;
        &lt;Dialog.Content&gt;
          &lt;TextInput
            autoCapitalize=&quot;none&quot;
            placeholderTextColor=&quot;rgba(0, 0, 0, 0.5)&quot;
            onChangeText={value =&gt; {
              setName(value);
            }}
            style={styles.inputeFileName}
            value={name}
            placeholder={dialogPlaceholder}
          /&gt;
        &lt;/Dialog.Content&gt;
        &lt;Dialog.Actions style={styles.btnRow}&gt;
          &lt;Button
            mode=&quot;outlined&quot;
            color=&quot;#003466&quot;
            uppercase={false}
            style={styles.btnStyle}
            labelStyle={styles.btnText}
            onPress={hide}
          &gt;
            {i18n.t(&apos;button.cancel&apos;)}
          &lt;/Button&gt;
          &lt;Button
            mode=&quot;contained&quot;
            uppercase={false}
            color=&quot;#003466&quot;
            labelStyle={styles.btnText}
            onPress={create}
          &gt;
            {i18n.t(&apos;button.create&apos;)}
          &lt;/Button&gt;
        &lt;/Dialog.Actions&gt;
      &lt;/Dialog&gt;
    &lt;/Portal&gt;
  );
};

/**
 * Styles of OwnSpaceDialog component
 */
const styles = StyleSheet.create({
  inputeFileName: {
    backgroundColor: &apos;rgba(255, 255, 255, 0.06)&apos;,
    borderRadius: 4,
    paddingLeft: 14,
    paddingTop: 10,
    paddingBottom: 10,
    borderWidth: 1,
    borderColor: &apos;#ccc&apos;,
    color: &apos;#555&apos;,
    fontSize: 16,
    height: 44
  },
  btnText: { fontSize: 16, fontFamily: &apos;DejaVuSans&apos; },
  btnStyle: { borderColor: &apos;#003466&apos; },
  btnRow: {
    justifyContent: &apos;space-between&apos;,
    marginHorizontal: 16,
    marginBottom: 12
  }
});

OwnSpaceDialog.propTypes = {
  name: PropTypes.string.isRequired,
  visible: PropTypes.bool.isRequired,
  hide: PropTypes.func.isRequired,
  setName: PropTypes.func.isRequired,
  create: PropTypes.func.isRequired
};

export default OwnSpaceDialog;
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
