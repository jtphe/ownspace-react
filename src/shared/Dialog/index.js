import React from 'react';
import { TextInput, StyleSheet } from 'react-native';
import Text from '@shared/ClientText';
import { Portal, Dialog, Button } from 'react-native-paper';
import i18n from '@i18n/i18n';
import PropTypes from 'prop-types';

/**
 * The OwnSpaceDialog component
 * @param {boolean} visible - Visibility of the dialog
 * @param {function} hide - Hide the dialog onDismiss
 * @param {string} dialogTitle - Text for the dialog title
 * @param {string} dialogPlaceholder - Text for the dialog placeholder
 * @param {string} name - Name
 * @param {function} setName - Set the name
 * @param {function} create - Start the function associated to the create button
 */
const OwnSpaceDialog = ({
  visible,
  hide,
  dialogTitle,
  dialogPlaceholder,
  name,
  setName,
  create
}) => {
  /**
   * Render the OwnSpaceDialog component
   * @returns {React.Component} - OwnSpaceDialog component
   */
  return (
    <Portal>
      <Dialog visible={visible} onDismiss={hide}>
        <Dialog.Title>
          <Text>{dialogTitle}</Text>
        </Dialog.Title>
        <Dialog.Content>
          <TextInput
            autoCapitalize="none"
            placeholderTextColor="rgba(0, 0, 0, 0.5)"
            onChangeText={value => {
              setName(value);
            }}
            style={styles.inputeFileName}
            value={name}
            placeholder={dialogPlaceholder}
          />
        </Dialog.Content>
        <Dialog.Actions style={styles.btnRow}>
          <Button
            mode="outlined"
            color="#003466"
            uppercase={false}
            style={styles.btnStyle}
            labelStyle={styles.btnText}
            onPress={hide}
          >
            {i18n.t('button.cancel')}
          </Button>
          <Button
            mode="contained"
            uppercase={false}
            color="#003466"
            labelStyle={styles.btnText}
            onPress={create}
          >
            {i18n.t('button.create')}
          </Button>
        </Dialog.Actions>
      </Dialog>
    </Portal>
  );
};

/**
 * Styles of OwnSpaceDialog component
 */
const styles = StyleSheet.create({
  inputeFileName: {
    backgroundColor: 'rgba(255, 255, 255, 0.06)',
    borderRadius: 4,
    paddingLeft: 14,
    paddingTop: 10,
    paddingBottom: 10,
    borderWidth: 1,
    borderColor: '#ccc',
    color: '#555',
    fontSize: 16,
    height: 44
  },
  btnText: { fontSize: 16, fontFamily: 'DejaVuSans' },
  btnStyle: { borderColor: '#003466' },
  btnRow: {
    justifyContent: 'space-between',
    marginHorizontal: 16,
    marginBottom: 12
  }
});

OwnSpaceDialog.propTypes = {
  name: PropTypes.string.isRequired,
  visible: PropTypes.bool.isRequired,
  hide: PropTypes.func.isRequired,
  setName: PropTypes.func.isRequired,
  create: PropTypes.func.isRequired
};

export default OwnSpaceDialog;
