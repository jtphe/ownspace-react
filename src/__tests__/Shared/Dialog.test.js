import 'react-native';
import React from 'react';
import Dialog from '@shared/Dialog';
import renderer from 'react-test-renderer';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Dialog tests', () => {
  it('Dialog snapshot', () => {
    const tree = renderer.create(<Dialog visible={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

Enzyme.configure({ adapter: new Adapter() });
