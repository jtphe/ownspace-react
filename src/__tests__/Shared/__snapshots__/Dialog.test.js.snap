// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog tests Dialog snapshot 1`] = `null`;
